tools/src/test/java/org/apache/kafka/tools/consumer/group/ConsumerRunnable.java

```diff
@@ -0,0 +1,113 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.tools.consumer.group;
+
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.errors.WakeupException;
+import org.apache.kafka.common.serialization.StringDeserializer;
+
+import java.time.Duration;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Properties;
+
+import static java.util.Collections.singleton;
+import static org.apache.kafka.clients.consumer.ConsumerConfig.GROUP_PROTOCOL_CONFIG;
+import static org.apache.kafka.clients.consumer.ConsumerConfig.GROUP_REMOTE_ASSIGNOR_CONFIG;
+import static org.apache.kafka.common.GroupType.CONSUMER;
+
+public class ConsumerRunnable implements Runnable {
+    final String broker;
+    final String groupId;
+    final Optional<Properties> customPropsOpt;
+    final boolean syncCommit;
+    final String topic;
+    final String groupProtocol;
+    final String assignmentStrategy;
+    final Optional<String> remoteAssignor;
+    final Properties props = new Properties();
+    volatile boolean isShutdown = false;
+    KafkaConsumer<String, String> consumer;
+
+    boolean configured = false;
+
+    public ConsumerRunnable(String broker,
+                            String groupId,
+                            String groupProtocol,
+                            String topic,
+                            String assignmentStrategy,
+                            Optional<String> remoteAssignor,
+                            Optional<Properties> customPropsOpt,
+                            boolean syncCommit) {
+        this.broker = broker;
+        this.groupId = groupId;
+        this.customPropsOpt = customPropsOpt;
+        this.syncCommit = syncCommit;
+
+        this.topic = topic;
+        this.groupProtocol = groupProtocol;
+        this.assignmentStrategy = assignmentStrategy;
+        this.remoteAssignor = remoteAssignor;
+    }
+
+    void configure() {
+        configured = true;
+        configure(props);
+        customPropsOpt.ifPresent(props::putAll);
+        consumer = new KafkaConsumer<>(props);
+    }
+
+    void configure(Properties props) {
+        props.put("bootstrap.servers", broker);
+        props.put("group.id", groupId);
+        props.put("key.deserializer", StringDeserializer.class.getName());
+        props.put("value.deserializer", StringDeserializer.class.getName());
+        props.put(GROUP_PROTOCOL_CONFIG, groupProtocol);
+        if (Objects.equals(groupProtocol, CONSUMER.toString())) {
+            remoteAssignor.ifPresent(assignor -> props.put(GROUP_REMOTE_ASSIGNOR_CONFIG, assignor));
+        } else {
+            props.put("partition.assignment.strategy", assignmentStrategy);
+        }
+    }
+
+    void subscribe() {
+        consumer.subscribe(singleton(topic));
+    }
+
+    @Override
+    public void run() {
+        assert configured : "Must call configure before use";
```

Comment:
We expect `configure` must be called so it would be nice to move it to the constructor. It means we should create the consumer in constructor.